libdevcrypto/OverlayDB.cpp

```diff
@@ -32,11 +32,11 @@ namespace dev
 
 h256 const EmptyTrie = sha3(rlp(""));
 
-OverlayDB::~OverlayDB()
-{
-	if (m_db.use_count() == 1 && m_db.get())
-		cnote << "Closing state DB";
-}
+#ifdef PRUNING
+u256 OverlayDB::m_blockNumber = 0;
+std::map<u256, std::set<h256> > OverlayDB::m_deathrow = {};
```

Comment:
Do these really need to be static? If so, does it require thread synchronization on access?, also consider using unordered_map/unordered_set if order is not important